test/spec/Feature/Query/UpdateSpec.hs

```diff
@@ -514,102 +477,56 @@ spec = do
-          ]|]
+    it "works with views with an explicit order by unique col" $
+      verifyMutation "limited_update_items_view" tblDataBefore
+        [json|[
+          { "id": 1, "name": "item-1" }
+        , { "id": 2, "name": "updated-item" }
+        , { "id": 3, "name": "item-3" }
+        ]|]
+        $
+        request methodPatch "/limited_update_items_view?order=id&limit=1&offset=1"
+            [("Prefer", "tx=commit")]
+            [json| {"name": "updated-item"} |]
+          `shouldRespondWith`
+            ""
+            { matchStatus  = 204
+            , matchHeaders = [ matchHeaderAbsent hContentType
+                             , "Preference-Applied" <:> "tx=commit" ]
+            }
 
-      request methodPatch "/limited_update_items_no_pk?order=ctid&limit=1"
-          [("Prefer", "tx=commit")]
-          [json| {"name": "updated-item"} |]
-        `shouldRespondWith`
-          ""
-          { matchStatus  = 204
-          , matchHeaders = [ matchHeaderAbsent hContentType
-                           , "Preference-Applied" <:> "tx=commit" ]
-          }
+    it "works with views with an explicit order by composite pk" $
+      verifyMutation "limited_update_items_cpk_view" tblDataBefore
+        [json|[
+          { "id": 1, "name": "item-1" }
+        , { "id": 2, "name": "updated-item" }
+        , { "id": 3, "name": "item-3" }
+        ]|]
+        $
+        request methodPatch "/limited_update_items_cpk_view?order=id,name&limit=1&offset=1"
+            [("Prefer", "tx=commit")]
+            [json| {"name": "updated-item"} |]
+          `shouldRespondWith`
+            ""
+            { matchStatus  = 204
+            , matchHeaders = [ matchHeaderAbsent hContentType
+                             , "Preference-Applied" <:> "tx=commit" ]
+            }
 
-      get "/limited_update_items_no_pk?order=id"
-        `shouldRespondWith`
-          [json|[
-            { "id": 1, "name": "updated-item" }
-          , { "id": 2, "name": "item-2" }
-          , { "id": 3, "name": "item-3" }
-          ]|]
-
-      request methodPost "/rpc/reset_items_tables"
-        [("Prefer", "tx=commit")]
-        [json| {"tbl_name": "limited_update_items_no_pk"} |]
-        `shouldRespondWith` ""
-        { matchStatus  = 204 }
+    it "works on a table without a pk by ordering by 'ctid'" $
+      verifyMutation "limited_update_items_no_pk" tblDataBefore
+        [json|[
+          { "id": 1, "name": "updated-item" }
+        , { "id": 2, "name": "item-2" }
+        , { "id": 3, "name": "item-3" }
+        ]|]
+        $
+        request methodPatch "/limited_update_items_no_pk?order=ctid&limit=1"
+            [("Prefer", "tx=commit")]
+            [json| {"name": "updated-item"} |]
+          `shouldRespondWith`
+            ""
+            { matchStatus  = 204
+            , matchHeaders = [ matchHeaderAbsent hContentType
+                             , "Preference-Applied" <:> "tx=commit" ]
+            }
```

Comment:
With the above improvements it could look like this:
```haskell
    it "works on a table without a pk by ordering by 'ctid'" $
      verifyMutation "limited_update_items_no_pk?order=id" tblDataBefore
        methodPatch "/limited_update_items_no_pk?order=ctid&limit=1" [json| {"name": "updated-item"} |] $
        [json|[
          { "id": 1, "name": "updated-item" }
        , { "id": 2, "name": "item-2" }
        , { "id": 3, "name": "item-3" }
        ]|]
```

Reply:
Also an idea, since there are 3 steps to this check, perhaps we can use [infix operators](https://stackoverflow.com/questions/6354401/anyone-know-how-to-use-a-partially-applied-three-argument-function-infix-haskel) to make it clearer. Like:

```haskell
"limited_update_items_no_pk?order=id" tblDataBefore `mutationBefore` 
methodPatch "/limited_update_items_no_pk?order=ctid&limit=1" `mutationAfter`        
[json|[
          { "id": 1, "name": "updated-item" }
        , { "id": 2, "name": "item-2" }
        , { "id": 3, "name": "item-3" }
        ]|]
```